### ISSUE TITLE
Add L2 normalization

### ISSUE THREAD

**Issue by JohnGiorgi** [JohnGiorgi/t2t#33]
WIP.

**Comment by JohnGiorgi** [comment]
This appears to be happening in the PyTorch Metric Learning loss functions by default (see [here](https://github.com/KevinMusgrave/pytorch-metric-learning/blob/f0be0fa2c6987c64de4a09375c731d976e10e953/pytorch_metric_learning/losses/base_metric_loss_function.py#L49). Closing for now, can re-open if I am mistaken.
